.changeset/khaki-falcons-search.md

```diff
@@ -0,0 +1,5 @@
+---
+'@directus/storage-driver-supabase': patch
+---
+
+Fix Supabase upload failures by upgrading to v2.12.1 and adding proper error handling
```

Comment:
Changeset should be [written in past tense](https://directus.io/docs/community/contribution/pull-requests#changesets) for the changelog
 ```suggestion
Fixed Supabase storage driver file uploads and improved error handling
```

Reply:
Noted ;)